audio-device-client/explainer.md

```diff
@@ -0,0 +1,167 @@
+# Audio Device Client: Better and Faster Audio I/O on Web
+
+This is a proposal for a new web API, `Audio Device Client`, which funcions as
+an intermediate layer between Web Audio API and actual audio devices used by the
+browser. It exposes various low-level properties that have been completely
+hidden or unavailable to developers.
+
+
+## Rationale
+
+Web Audio API has been criticized by the lack of ability to run low-level audio
+processing properly. W3C Audio Working Group addressed the problem with the
+newly added Audio Worklet, but it is still confined by the boundary of Web Audio
+API's graph rendering mechanism.
+
+To overcome this limitation, some "pro-audio" web apps were designed with a new
+processing model and it bypasses the most of Web Audio API's graph system by
+utilizing Audio Worklet, SharedArrayBuffer and Worker.
+
+![Design pattern: WebAudio Powerhouse](https://github.com/WebAudio/web-audio-cg/blob/master/audio-device-client/images/img-1-design-pattern.png "Design pattern: WebAudio Powerhouse")
+
+This setup is convenient to take advantage of the code compiled into
+WebAssembly, therefore this new model cuts the engineering cost significantly
+because audio developers can bring existing source codes to the web platform
+with minimum effort. Additionally, deploying to multiple targets by using same
+source code ensures the identical sonic result across various platforms.
+(e.g. encoding/decoding, signal processing or an entire audio application)
+
+However, this convoluted workaround only solves some of problems that this breed
+of audio-centric applications faces. It is still locked by the graph render
+quantum size (128 sample-frames), and there is no unified API that configures
+essential properties of audio rendering mechanism (such as I/O device selection,
+multi-channel I/O support, configurable sample rate and more).
+
+The slow adoption of Web Audio API from pro-audio or game industry is the clear
+evidence of why this problem is important to solve.
```

Comment:
Not really clear what "this problem" is.  The previous paragraphs are kind of mixed in old and new styles.  Maybe you should focus on how it has to be done today and explain why it's a pain.

Reply:
I will think about how to expand this a bit more.